Content/Items/Magnet/Accessories.WatchBattery.cs

```diff
@@ -0,0 +1,53 @@
+﻿using StarlightRiver.Content.Items.BaseTypes;
+using StarlightRiver.Content.Items.Manabonds;
+using StarlightRiver.Helpers;
+using Terraria.ID;
+
+namespace StarlightRiver.Content.Items.Magnet
+{
+	internal class WatchBattery : SmartAccessory
+	{
+		public override string Texture => AssetDirectory.MagnetItem + Name;
+
+		public WatchBattery() : base("Watch Battery", "Your sentries shock nearby enemies when placed") { }
+
+		public override void Load()
+		{
+			StarlightProjectile.PostAIEvent += SpawnShock;
+		}
+
+		public override void SafeSetDefaults()
+		{
+			Item.rare = ItemRarityID.Blue;
+			Item.value = Item.sellPrice(silver: 50);
+		}
+
+		private void SpawnShock(Projectile projectile)
+		{
+			Player owner = Main.player[projectile.owner];
+
+			if (owner != null && Equipped(owner))
```

Comment:
this whole block could check against local player from what I see to make sure other clients don't execute this